File: backend/campusbites-backend/src/main/java/com/campusbites/product/model/Product.java
package com.campusbites.product.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.*;
import org.springframework.data.mongodb.core.mapping.Document;

import java.math.BigDecimal;
import java.time.Instant;

@Data
@NoArgsConstructor
@Document(collection = "products")
public class Product {
    @Id private String id;
    private String     name;
    private String     description;
    private String     category;
    private BigDecimal price;
    private String     imageUrl;
    private double     rating      = 0.0;
    private int        ratingCount = 0;
    private boolean    available   = true;
    private int        stock       = 0;
    @CreatedDate      private Instant createdAt;
    @LastModifiedDate private Instant updatedAt;
}

File: mobile/app/src/main/java/com/example/campusbites/features/home/data/model/Product.kt
package com.example.campusbites.features.home.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Product(
    @SerialName("_id") val id: String = "",
    val name: String                  = "",
    val category: String              = "",
    val price: Double                 = 0.0,
    val description: String           = "",
    val image: String?                = null,
    val rating: Double                = 4.5,
)

@Serializable
data class ProductPageResponse(
    val data: ProductData? = null,
)

@Serializable
data class ProductData(
    val content: List<Product>? = null,
)